### ISSUE TITLE
`items_after_statements` doesn't understand `cfg_select!`

### ISSUE THREAD

**Issue by nazar-pc** [rust-lang/rust-clippy#17498]
### Summary

The latest nightly introduced formatting support for `cfg_select!`, which led me to remove a bunch of extra `{}` in conditional blocks. Unfortunately this started triggering this lint unnecessarily.

### Lint Name

items_after_statements

### Reproducer

I tried this code:

```rust
cfg_select! {
    unix => {
        use std::os::unix::fs::FileExt;

        self.file.read_exact_at(buf, offset)
    }
    windows => {
        use std::os::windows::fs::FileExt;

        // ...
```

I saw this happen:

```
warning: adding items after statements is confusing, since items exist from the start of the scope
   --> crates/shared/ab-direct-io-file/src/lib.rs:384:17
    |
384 |                 use std::os::unix::fs::FileExt;
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#items_after_statements
    = note: `-W clippy::items-after-statements` implied by `-W clippy::pedantic`
    = help: to override `-W clippy::pedantic` add `#[allow(clippy::items_after_statements)]`

warning: adding items after statements is confusing, since items exist from the start of the scope
   --> crates/shared/ab-direct-io-file/src/lib.rs:436:17
    |
436 |                 use std::os::unix::fs::FileExt;
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#items_after_statements
```

Adding extra `{}` helps, but `rustfmt` removes them automatically:
```rust
cfg_select! {
    unix => {{
        use std::os::unix::fs::FileExt;

        self.file.read_exact_at(buf, offset)
    }}
    windows => {{
        use std::os::windows::fs::FileExt;

        // ...
```

I expected to see this happen:
No warnings

### Version

```text
rustc 1.99.0-nightly (11177f223 2026-08-02)
binary: rustc
commit-hash: 11177f2235f0c842b00f82c558ad9480c0c3a895
commit-date: 2026-08-02
host: x86_64-unknown-linux-gnu
release: 1.99.0-nightly
LLVM version: 22.1.8
```

**Comment by blyxyas** [comment]
Hi @nazar-pc , I assume that this happens in some structure [like this](https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&code=%23%21%5Bwarn%28clippy%3A%3Aitems_after_statements%29%5D%0A%23%21%5Ballow%28unused%29%5D%0Ause+std%3A%3Afs%3A%3AFile%3B%0Astruct+S+%7B%0A++++file%3A+File%2C%0A%7D%0A%0Aimpl+S+%7B%0A++++fn+s%28self%29+%7B%0A++++++++let+buf+%3D+%26mut+%5B15%3B+1%5D%3B%0A++++++++let+offset+%3D+8%3B%0A++++++++cfg_select%21+%7B%0A++++++++++++unix+%3D%3E+%7B%0A++++++++++++++++use+std%3A%3Aos%3A%3Aunix%3A%3Afs%3A%3AFileExt%3B%0A%0A++++++++++++++++let+_+%3D+self.file.read_exact_at%28buf%2C+offset%29%3B%0A++++++++++++%7D%0A++++++++++++windows+%3D%3E+%7B%0A++++++++++++++++use+std%3A%3Aos%3A%3Awindows%3A%3Afs%3A%3AFileExt%3B%0A++++++++%7D%7D%0A++++++++%0A++++%7D%0A%7D%0A), right?


```
#![warn(clippy::items_after_statements)]
#![allow(unused)]
use std::fs::File;
struct S {
    file: File,
}

impl S {
    fn s(self) {
        let buf = &mut [15; 1];
        let offset = 8;
        cfg_select! {
            unix => {
                use std::os::unix::fs::FileExt;

                let _ = self.file.read_exact_at(buf, offset);
            }
            windows => {
                use std::os::windows::fs::FileExt;
        }}
        
    }
}

```

**Comment by nazar-pc** [comment]
Yes, here is example in a larger project: https://github.com/nazar-pc/abundance/blob/323942848c6dfab14076438adaead9f49aa13ae9/crates/shared/ab-direct-io-file/src/lib.rs#L389-L430

**Comment by blyxyas** [comment]
Okay, so this is caused because `cfg_select! doesn't seem to mark itself as a macro, which I think is a bug In `cfg_select` itself. I'll open an issue/PR upstream, as this is not an issue with Clippy.
